pkg/server/util/regentryutil/fetch.go

```diff
@@ -3,16 +3,19 @@ package regentryutil
 import (
 	"context"
 	"errors"
+	"time"
 
 	"github.com/golang/protobuf/ptypes/wrappers"
 	"github.com/spiffe/spire/pkg/common/util"
 	"github.com/spiffe/spire/pkg/server/plugin/datastore"
 	"github.com/spiffe/spire/proto/spire/common"
 )
 
-func FetchRegistrationEntries(ctx context.Context, dataStore datastore.DataStore, spiffeID string) ([]*common.RegistrationEntry, error) {
+const cacheFetchEntriesTTL = 1 * time.Second
+
+func FetchRegistrationEntries(ctx context.Context, dataStore datastore.DataStore, cache RegistrationEntriesCache, spiffeID string) ([]*common.RegistrationEntry, error) {
```

Comment:
I'm sorry that I wasn't more clear in my previous comment. I still think there is benefit to having `FetchRegistrationEntries` and `FetchRegistrationEntriesWithCache` variants. My comment about the "cache" interface was more for internal usage, i.e. `FetchRegistrationEntries` passes the `nullCache` to the fetcher, `FetchRegistrationEntriesWithCache` passes the provided cache implementation to the fetcher.

Reply:
gotcha!